### PR TITLE
Cake baking autotest

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

I want to get back to the food stuff, and adding a few auto tests to avoid regressions.

The test makes a dummy human bake a cake from the base ingredients, confirming that the reagent purities are correct and that the cake slice gives a food buff in the end.

## Why It's Good For The Game

Avoiding bugs.

### CODE REVIEW

**Comment by Mothblocks** [comment]
the term is unit test

File: code/modules/unit_tests/bake_a_cake.dm [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	var/egg_purity = sample_egg.reagents.get_average_purity()
+	TEST_ASSERT_EQUAL(egg_purity, CONSUMABLE_STANDARD_PURITY, "Expected starting purity of egg reagents to be [CONSUMABLE_STANDARD_PURITY], but got [egg_purity]!")
+
+	human.mind = new /datum/mind(null) // Add brain for the food buff
+
+	// It's a piece of cake to bake a pretty cake
+	sugar_bag.melee_attack_chain(human, beaker)
+	for(var/i in 1 to 3)
+		flour_bag.melee_attack_chain(human, beaker)
+	TEST_ASSERT_EQUAL(beaker.reagents.total_volume, 20, "Failed to pour sugar & flour for the cake bater!")
```
Comment (Mothblocks) [inline code/modules/unit_tests/bake_a_cake.dm:31]: batter

File: code/modules/unit_tests/bake_a_cake.dm [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	for(var/i in 1 to 3)
+		var/obj/item/egg = egg_box.contents[1]
+		egg.melee_attack_chain(human, beaker, RIGHT_CLICK)
+	var/obj/item/food/cake_bater = locate(/obj/item/food/cakebatter) in table_loc
```
Comment (Mothblocks) [inline code/modules/unit_tests/bake_a_cake.dm:36]: the word is batter

**Review by Mothblocks — APPROVED** [review]

File: code/modules/unit_tests/bake_a_cake.dm [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	var/obj/structure/table/the_table = allocate(/obj/structure/table, table_loc)
+	var/obj/item/knife/kitchen/a_knife = allocate(/obj/item/knife/kitchen, table_loc)
+	var/obj/item/reagent_containers/cup/beaker/beaker = allocate(/obj/item/reagent_containers/cup/beaker, table_loc)
+	var/obj/item/reagent_containers/condiment/flour/flour_bag = allocate(/obj/item/reagent_containers/condiment/flour, table_loc)
+	var/obj/item/reagent_containers/condiment/sugar/sugar_bag = allocate(/obj/item/reagent_containers/condiment/sugar, table_loc)
+	var/obj/item/storage/fancy/egg_box/egg_box = allocate(/obj/item/storage/fancy/egg_box, table_loc)
+	var/obj/item/food/egg/sample_egg = egg_box.contents[1]
+
+	var/sugar_purity = sugar_bag.reagents.get_average_purity()
+	TEST_ASSERT_EQUAL(sugar_purity, 1, "Expected starting purity of a default sugar to be [1], but got [sugar_purity]!")
```
Comment (Ghommie) [inline code/modules/unit_tests/bake_a_cake.dm:19]:
```suggestion
	TEST_ASSERT_EQUAL(sugar_purity, 1, "Expected starting purity of a default sugar to be 1, but got [sugar_purity]!")
```
Unless you're making that 1 purity a define, it isn't necessary.

File: code/modules/unit_tests/bake_a_cake.dm [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+	for(var/i in 1 to 3)
+		var/obj/item/egg = egg_box.contents[1]
+		egg.melee_attack_chain(human, beaker, RIGHT_CLICK)
+	var/obj/item/food/cake_batter = locate(/obj/item/food/cakebatter) in table_loc
+	TEST_ASSERT_NOTNULL(cake_batter, "Failed making cake batter!")
+	TEST_ASSERT_EQUAL(beaker.reagents.total_volume, 0, "Cake batter did not consume all beaker reagents!")
+
+	var/batter_purity = cake_batter.reagents.get_average_purity()
+	var/batter_purity_expected = (5 * sugar_purity + 15 * flour_purity + 18 * egg_purity) / 38
```
Comment (Ghommie) [inline code/modules/unit_tests/bake_a_cake.dm:41]: What do the numbers 5, 15, 18 and 38 represent?

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/bake_a_cake.dm (hunk 4 above)
Comment (MTandi) [reply code/modules/unit_tests/bake_a_cake.dm:41]: Volumes. Can grab that from the recipe I guess

**Review by Ghommie — APPROVED** [review]

**Pull request merged**